crates/bin/node/src/grpc_service.rs

```diff
@@ -554,6 +555,31 @@ impl Node for GrpcState {
         Ok(Response::new(AcknowledgeResponse {}))
     }
 
+    async fn check_state(
+        &self,
+        request: Request<CheckStateRequest>,
+    ) -> Result<Response<CheckStateResponse>, Status> {
+        let request = request.into_inner();
+
+        let ys = request.ys;
+
+        let proof_state = self
+            .inner_check_state(ys)
+            .await
+            .map_err(|e| Status::internal(format!("Failed to check state: {}", e)))?
```

Comment:
rather than this `impl From<Error> for Status` on your error type in check_state/mod.rs

Reply:
DbConnection -> Internal
ProofStateRetrieval -> not found